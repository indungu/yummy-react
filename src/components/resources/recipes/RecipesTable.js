import React, { Component } from 'react';
import { Table } from 'react-bootstrap';

export default class RecipesTable extends Component {
  constructor(props) {
    super(props);
    this.state = {
      toEdit: '',
      toDelete: '',
    };
  }
  render() {
    if (this.props.recipes.length !== 0 && this.props.categoryId !== undefined) {
      console.log(this.props);
      return (
        <div>
          <AddRecipeButton
            categoryName={this.props.categoryName}
            categoryId={this.props.categoryId}
            onAddRecipeClick={this.props.onAddRecipeClick}
          />
          <Table>
            <thead>
              <tr>
                <th>Name</th>
                <th>Ingredients</th>
                <th>Description</th>
                <th>Action</th>
              </tr>
            </thead>
            <tbody>
              { this.props.data.map((data, index) => {
                return (
                  <tr key={index} >
                    <td>{ data.recipe_name }</td>
                    <td>{ data.recipe_ingredients }</td>
                    <td>{ data.recipe_description }</td>
                    <td>
                      <div className="btn-group">
                        <button
                          className="btn btn-primary btn-sm"
                          id={data.recipe_id}
                          name={data.recipe_name}
                          onClick={this.props.listRecipes}
                          style={{ marginRight: '5px' }}
                        >View Recipe
                        </button>
                        <button
                          className="btn btn-success btn-sm"
                          name={data.recipe_name}
                          onClick={this.launchEditPrompt}
                          data-toggle="modal"
                          // eslint-disable-next-line
                          data-target={'#'+data.recipe_name}
                          style={{ marginRight: '5px' }}
                        >Edit recipe
                        </button>
                        <button
                          className="btn btn-danger btn-sm"
                          name={data.recipe_name}
                          onClick={this.launchDeletePrompt}
                          data-toggle="modal"
                          // eslint-disable-next-line
                          data-target={'#'+data.recipe_name}
                        >Delete recipe
                        </button>
                      </div>
                    </td>
                  </tr>
                );
              })}
            </tbody>
          </Table>
        </div>
      );
    } else {
      return (
        <div className="" style={{ display: this.props.display }}>
          <AddRecipeButton
            categoryName={this.props.categoryName}
            categoryId={this.props.categoryId}
            onAddRecipeClick={this.props.onAddRecipeClick}
          />
          <p className="text-center info">
            You currently have no recipes for category {this.props.categoryName}.
          </p>
        </div>
      );
    }
  }
}

const AddRecipeButton = (props) => {
  console.log(`child props ${props.onAddRecipeClick}`);
  return (
    <div className="col-md-2 add-recipe">
      <div className="well">
        <button
          className="btn btn-outline-primary btn-block"
          name={props.categoryName}
          id={props.categoryId}
          onClick={props.onAddRecipeClick}
        >Add Recipe
        </button>
      </div>
    </div>
  );
};
