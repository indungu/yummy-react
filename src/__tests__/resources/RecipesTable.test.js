import React from 'react';
import { mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import RecipesTable from '../../components/resources/recipes/RecipesTable';

describe('RecipesTable', () => {
  const props = {
    data: [{
      name: 'some cookie',
      ingredients: 'some ingredient',
      description: 'some decription',
    }],
    categoryId: 1,
  };
  const wrapper = mount(
    <RecipesTable
      data={props.data}
      categoryId={props.categoryId}
    />,
  );
  it('Renders properly ', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
