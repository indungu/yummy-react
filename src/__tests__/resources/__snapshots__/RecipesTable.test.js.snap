// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipesTable Renders properly  1`] = `
<Component
  categoryId={1}
  data={
    Array [
      Object {
        "description": "some decription",
        "ingredients": "some ingredient",
        "name": "some cookie",
      },
    ]
  }
>
  <div>
    <div
      className="row"
    >
      <AddRecipeButton
        categoryId={1}
      >
        <div
          className="col-md-2"
        >
          <div
            className="well"
          >
            <button
              className="btn btn-outline-primary btn-block"
              id={1}
            >
              Add Recipe
            </button>
          </div>
        </div>
      </AddRecipeButton>
      <Component
        scope="Recipe Name"
      >
        <div
          className="col-md-10"
        >
          <form
            className="form-row align-items-center"
          >
            <div
              className="form-group mx-sm-3 mb-2 col-9"
            >
              <input
                className="form-control"
                name="queryString"
                placeholder="Recipe Name"
                type="text"
              />
            </div>
            <button
              className="btn btn-primary mb-2"
              type="submit"
            >
              Search
            </button>
          </form>
        </div>
      </Component>
    </div>
    <Table
      bordered={false}
      bsClass="table"
      condensed={false}
      hover={false}
      responsive={false}
      striped={false}
    >
      <table
        className="table"
      >
        <thead>
          <tr>
            <th>
              Name
            </th>
            <th>
              Ingredients
            </th>
            <th>
              Description
            </th>
            <th>
              Action
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            key="0"
          >
            <td>
              some cookie
            </td>
            <td>
              some ingredient
            </td>
            <td>
              some decription
            </td>
            <td>
              <div
                className="btn-group"
              >
                <button
                  className="btn btn-success btn-sm"
                  data-target="#editsome cookie"
                  data-toggle="modal"
                  name="some cookie"
                  style={
                    Object {
                      "marginRight": "5px",
                    }
                  }
                >
                  Edit Recipe
                </button>
                <button
                  className="btn btn-danger btn-sm"
                  data-target="#some cookie"
                  data-toggle="modal"
                  name="some cookie"
                >
                  Delete Recipe
                </button>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </Table>
    <Component>
      <nav
        aria-label="Page navigation"
      >
        <ul
          className="pagination justify-content-center"
        />
      </nav>
    </Component>
  </div>
</Component>
`;
